src/overlays/actors/ovl_En_Horse/z_en_horse.c

```diff
@@ -90,253 +139,3541 @@ static ColliderJntSphInit D_80A65F28 = {
+    Vec3s* jointTable;
+    float y;
+
+    this->action = ENHORSE_ACT_MOUNTED_GALLOP;
+    this->animationIdx = ENHORSE_ANIM_GALLOP;
+    Animation_PlayOnce(&this->skin.skelAnime, sAnimationHeaders[this->type][this->animationIdx]);
+    jointTable = this->skin.skelAnime.jointTable;
+    y = jointTable->y;
+    this->riderPos.y += (y * 0.01f);
+    this->postDrawFunc = NULL;
+}
+
+void EnHorse_StartBraking(EnHorse* this, GlobalContext* globalCtx);
+void EnHorse_MountedGallop(EnHorse* this, GlobalContext* globalCtx) {
+    f32 stickMag;
+    s16 stickAngle;
+
+    EnHorse_StickDirection(&this->curStick, &stickMag, &stickAngle);
+
+    if (this->noInputTimer <= 0.0f) {
+        EnHorse_UpdateSpeed(this, globalCtx, 0.3f, -0.5f, 10.0f, 0.06f, 8.0f, 0x190);
+    } else if (this->noInputTimer > 0.0f) {
+        this->noInputTimer -= 1;
+        this->actor.speedXZ = 8.0f;
+    }
+    if (this->actor.speedXZ < 6.0f) {
+        EnHorse_StartTrotting(this);
+    }
+
+    this->skin.skelAnime.playSpeed = this->actor.speedXZ * 0.3f;
+    if (SkelAnime_Update(&this->skin.skelAnime)) {
+        EnHorse_PlayGallopingSound(this);
+        func_800AA000(0, 120, 8, 255);
+        if (EnHorse_PlayerCanMove(this, globalCtx) == 1) {
+            if (stickMag >= 10.0f && Math_CosS(stickAngle) <= -0.5f) {
+                EnHorse_StartBraking(this, globalCtx);
+            } else if (this->actor.speedXZ < 6.0f) {
+                EnHorse_StartTrotting(this);
+            } else {
+                EnHorse_MountedGallopReset(this);
+            }
+            return;
+        }
+        EnHorse_MountedGallopReset(this);
+    }
+}
+
+void EnHorse_StartRearing(EnHorse* this) {
+    this->action = ENHORSE_ACT_MOUNTED_REARING;
+    this->animationIdx = ENHORSE_ANIM_REARING;
+    this->flags &= ~ENHORSE_LAND2_SOUND;
+    this->unk_21C = this->unk_228;
+    if (this->flags & ENHORSE_DRAW) {
+        Audio_PlaySoundGeneral(NA_SE_EV_HORSE_NEIGH, &this->unk_21C, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+    }
+    func_800AA000(0.0f, 180, 20, 100);
+    Animation_Change(&this->skin.skelAnime, sAnimationHeaders[this->type][this->animationIdx], 1.0f, 0.0f,
+                     Animation_GetLastFrame(sAnimationHeaders[this->type][this->animationIdx]), ANIMMODE_ONCE, -3.0f);
+}
+
+void EnHorse_MountedRearing(EnHorse* this, GlobalContext* globalCtx) {
+    f32 stickMag;
+    s16 stickAngle;
+
+    this->actor.speedXZ = 0;
+    if (this->curFrame > 25.0f) {
+        if (!(this->flags & ENHORSE_LAND2_SOUND)) {
+            this->flags |= ENHORSE_LAND2_SOUND;
+            Audio_PlaySoundGeneral(NA_SE_EV_HORSE_LAND2, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                                   &D_801333E8);
+            func_800AA000(0, 180, 20, 100);
+        }
+    }
+    EnHorse_StickDirection(&this->curStick, &stickMag, &stickAngle);
+    if (SkelAnime_Update(&this->skin.skelAnime) != 0) {
+        if (EnHorse_PlayerCanMove(this, globalCtx) == 1) {
```

Comment:
`SkelAnime_Update` returns a bool so you can do if (SkelAnime_Update), here and everywhere else.

unfortunate about the PlayerCanMove checking against 1 lol. can probably do if == true, we've been doing that in other places when this happens